samtranslator/model/__init__.py

```diff
@@ -536,6 +541,30 @@ def _construct_tag_list(
         # customer's knowledge.
         return get_tag_list(sam_tag) + get_tag_list(additional_tags) + get_tag_list(tags)
 
+    @staticmethod
+    def propagate_tags(
+        resources: List[Resource], tags: Optional[Dict[str, Any]], propagate_tags: Optional[bool] = False
+    ) -> None:
+        """
+        Propagates tags to the resources. This function assumes that generated resources always have
+        the tags property called `Tags`
+
+        TODO: There are CFN resources that do not follow the assumption such as 'AWS::Backup::Framework'
+        We may need to modify the code to account for thia scenario in the future.
```

Comment:
What is required to make it robust? Doesn't the current approach of overloading work with any situation?

Reply:
Yes, the comment is outdated. Will update it.